showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITClientShutdown.java

```diff
@@ -0,0 +1,188 @@
+import com.google.common.truth.Truth;
+import com.google.showcase.v1beta1.BlockRequest;
+import com.google.showcase.v1beta1.BlockResponse;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoRequest;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import java.util.concurrent.TimeUnit;
+import org.junit.Test;
+import org.threeten.bp.Duration;
+
+public class ITClientShutdown {
+
+  // Test to ensure the client can close + terminate properly
+  @Test
+  public void testGrpc_closeClient() throws Exception {
+    EchoClient grpcClient = TestClientInitializer.createGrpcEchoClient();
+    grpcClient.close();
+    // 500ms buffer time to properly terminate the client
+    grpcClient.awaitTermination(500, TimeUnit.MILLISECONDS);
+    Truth.assertThat(grpcClient.isShutdown()).isTrue();
+    Truth.assertThat(grpcClient.isTerminated()).isTrue();
+  }
+
+  // Test to ensure the client can close + terminate properly
+  @Test
+  public void testHttpJson_closeClient() throws Exception {
+    EchoClient httpjsonClient = TestClientInitializer.createHttpJsonEchoClient();
+    httpjsonClient.close();
+    // 500ms buffer time to properly terminate the client
+    httpjsonClient.awaitTermination(500, TimeUnit.MILLISECONDS);
+    Truth.assertThat(httpjsonClient.isShutdown()).isTrue();
+    Truth.assertThat(httpjsonClient.isTerminated()).isTrue();
+  }
+
+  // Test to ensure hte client can close + terminate after a quick RPC invocation
+  @Test
+  public void testGrpc_rpcInvoked_closeClient() throws Exception {
+    EchoClient grpcClient = TestClientInitializer.createGrpcEchoClient();
+
+    grpcClient.echo(EchoRequest.newBuilder().setContent("Test").build());
+
+    grpcClient.close();
+    // 1s buffer time to properly terminate the client after RPC is invoked
+    grpcClient.awaitTermination(1, TimeUnit.SECONDS);
+    Truth.assertThat(grpcClient.isShutdown()).isTrue();
+    Truth.assertThat(grpcClient.isTerminated()).isTrue();
+  }
+
+  // Test to ensure hte client can close + terminate after a quick RPC invocation
+  @Test
+  public void testHttpJson_rpcInvoked_closeClient() throws Exception {
+    EchoClient httpjsonClient = TestClientInitializer.createHttpJsonEchoClient();
+
+    httpjsonClient.echo(EchoRequest.newBuilder().setContent("Test").build());
+
+    httpjsonClient.close();
+    // 1s buffer time to properly terminate the client after RPC is invoked
+    httpjsonClient.awaitTermination(1, TimeUnit.SECONDS);
+    Truth.assertThat(httpjsonClient.isShutdown()).isTrue();
+    Truth.assertThat(httpjsonClient.isTerminated()).isTrue();
+  }
+
+  // This test is to ensure that the client is able to close + terminate any resources
+  // once a response has been received. Set a max test duration of 15s to ensure that
+  // the test does not continue on forever.
+  @Test(timeout = 15000L)
+  public void testGrpc_rpcInvokedWithLargeTimeout_closeClientOnceResponseReceived()
```

Comment:
I think this test and the following one, `testHttpJson_rpcInvokedWithLargeTimeout_closeClientOnceResponseReceived`, are similar test with different transport. Could you combine them in a parameterized test?

Also, what test will be used to test `grpc+rest`? Sorry I'm new to this area.

Reply:
The tests are the same except for the transport. For the showcase tests, we've intentionally did not use parameterized tests for readability. IIRC, the reason being that we would be able to quickly read the test name and figure out what what is being tested. [ITIAM](https://github.com/googleapis/sdk-platform-java/blob/main/showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java) has the duplicated tests with the other difference being the transport. 

> Also, what test will be used to test grpc+rest? Sorry I'm new to this area.

I believe you're talking about the bazel rule input, right? `grpc+rest` tells the generator that generate grpc && httpjson stubs. You can't create a client with both transports at the same time (it's either gRPC or HttpJson).

Reply:
> For the showcase tests, we've intentionally did not use parameterized tests for readability.

Maybe duplicate test cases are fine in this PR since refactoring tests is not the main purpose of this PR.

However, I suspect the gain of (possible?) readability outrun the downside of code duplication. Imagine you need to modify *all* test to accommodate a code change. 

> I believe you're talking about the bazel rule input, right? grpc+rest tells the generator that generate grpc && httpjson stubs. You can't create a client with both transports at the same time (it's either gRPC or HttpJson).

Got it, makes sense.

Reply:
> However, I suspect the gain of (possible?) readability outrun the downside of code duplication. Imagine you need to modify all tests to accommodate a code change.

I understand the concern of having to make the same code change multiple times and ensuring that the changes all work. It was something brought up a while back when the showcase tests were first implemented and the preference was for readability and to not use parameterized tests.

Reply:
Do you think it makes sense to document the rational in https://github.com/googleapis/sdk-platform-java/blob/main/DEVELOPMENT.md#testing-guide in another PR (or create an issue)?

Reply:
Personally, I don't have a strong opinion regarding this since I see the pros/cons of both. I am currently leaning to just keep as-is and follow the precedence set in existing tests. If future tests in files are better suited one way, then we can make a case for it then.

Reply:
I don't think policy is set in stone for the showcase tests and I can create an issue to re-visit changing the tests to incorporate parameterized tests. It's been a while since the initial showcase tests were written and I think (?) there was a discussion a while back regarding using parameterized tests or duplicating tests for the transports.